### ISSUE TITLE
dock progress bar looks a bit dated

### ISSUE THREAD

**Issue by Feichtmeier** [ubuntu/yaru#4016]
we've talked about this some years ago :older_adult: :older_adult: :older_man: 

![image](https://github.com/ubuntu/yaru/assets/15329494/75e8e05c-d36f-41b9-9c5d-a195f61922ab)

IMO the progress bar look, engraved/3d + gradient track, does not really fit to modern gnome/yaru

@3v1n0 could you eventually change this to fit a bit better?

**Comment by Jupi007** [comment]
I fully agree on this.

**Comment by Feichtmeier** [comment]
<img width="361" alt="Bildschirmfoto 2023-11-20 um 15 42 00" src="https://github.com/ubuntu/yaru/assets/15329494/b8abcc98-bdef-45d0-983c-c6628527dcad">
this is from macos, which looks quite similar but doesnt have a gradient in trough

**Comment by Feichtmeier** [comment]
@vanvugt @sergio-costas since you touched the dock code recently, could you eventually check if you can fix this issue here?

**Comment by sergio-costas** [comment]
Is there a way of quickly emulate a bar?

**Comment by sergio-costas** [comment]
(to avoid having to do a big copy of files)

**Comment by sergio-costas** [comment]
Solved the emulation problem.

**Comment by Feichtmeier** [comment]
Thank you @sergio-costas 🙇 

I think removing the gradient and the highlight from the trough should be enough as the monochrome bar and background fits the rest of the shell. The bar could potenially be colored in the primary accent color (selection color) but since there are some risky colors in the settings (that super un saturated greens and browns...) this could not work.

**Comment by 3v1n0** [comment]
> Solved the emulation problem.

Oh, sorry... It can be easily done in python, eg:

```python
from gi.repository import Unity, Gio, GObject, Dbusmenu, GLib
import sys

loop = GObject.MainLoop()

# Pretend to be nautilus for the sake of the example 
launcher = Unity.LauncherEntry.get_for_desktop_id ("org.gnome.Nautilus.desktop")
nautilus_launcher = launcher
# launcher = Unity.LauncherEntry.get_for_desktop_id ("firefox.desktop")
#launcher = Unity.LauncherEntry.get_for_desktop_id ("org.gnome.Devhelp.desktop")

# Show a count on the icon
#launcher.set_property("count", 1234)
launcher.set_property("count_visible", True)
launcher.set_property("count", int(sys.argv[1]) if len(sys.argv) > 1 else 5)
Unity.LauncherEntry.get_for_desktop_id ("org.gnome.Calculator.desktop").set_property("urgent", True)

def on_timeout1():
    print('timeout1')
    nautilus_launcher.set_property("count", 5)
    return False

def on_timeout2():
    print('timeout2')
    nautilus_launcher.set_property("count", 12345)
    return False

def on_timeout3():
    print('timeout urgent off')
    Unity.LauncherEntry.get_for_desktop_id ("org.gnome.Calculator.desktop").set_property("urgent", False)
    return False

def on_timeout4():
    print('timeout urgent on')
    Unity.LauncherEntry.get_for_desktop_id ("org.gnome.Calculator.desktop").set_property("urgent", True)
    nautilus_launcher.set_property("count", 0)
    return False


GLib.timeout_add_seconds(3, on_timeout1)
GLib.timeout_add_seconds(5, on_timeout2)
GLib.timeout_add_seconds(6, on_timeout3)
GLib.timeout_add_seconds(7, on_timeout4)

# Set progress to 42% done 
launcher.set_property("progress", 0.42)
launcher.set_property("progress_visible", True)

# Set us as urgent. Quickly! Go! Go! Go! Now! Now!

launcher = Unity.LauncherEntry.get_for_desktop_id ("org.gnome.Terminal.desktop")
launcher.set_property("urgent", True)

def on_clicked(it, time):
	print ("Item clicked "+str(it)+" at "+str(time))

# We also want a quicklist 
ql = Dbusmenu.Menuitem.new ()
item1 = Dbusmenu.Menuitem.new ()
item1.property_set (Dbusmenu.MENUITEM_PROP_LABEL, "Item 1")
item1.property_set_bool (Dbusmenu.MENUITEM_PROP_VISIBLE, True)
item1.connect("item-activated", on_clicked)
item2 = Dbusmenu.Menuitem.new ()
item2.property_set (Dbusmenu.MENUITEM_PROP_LABEL, "Item 2")
item2.property_set_bool (Dbusmenu.MENUITEM_PROP_VISIBLE, True)
item2.connect("item-activated", on_clicked)
ql.child_append (item1)
ql.child_append (item2)
# item = Dbusmenu.Menuitem.new ()
# item.property_set (Dbusmenu.MENUITEM_PROP_LABEL, "Quit")
# item.property_set_bool (Dbusmenu.MENUITEM_PROP_VISIBLE, True)
# ql.child_append (item)
item = Dbusmenu.Menuitem.new ()
item.property_set (Dbusmenu.MENUITEM_PROP_LABEL, "Close This App")
item.property_set_bool (Dbusmenu.MENUITEM_PROP_VISIBLE, True)
item.property_set_bool ("unity-quit-action", True)
item.connect("item-activated", on_clicked)
ql.child_append (item)
launcher.set_property("quicklist", ql)

loop.run()
```

**Comment by sergio-costas** [comment]
@Feichtmeier Done. Tell me what do you think https://github.com/micheleg/dash-to-dock/pull/2153

@3v1n0 Ehm... I did it by adding `this.setProgress(0.4);` at the end of the constructor... 😅

**Comment by Feichtmeier** [comment]
> @Feichtmeier Done. Tell me what do you think [micheleg/dash-to-dock#2153](https://github.com/micheleg/dash-to-dock/pull/2153)
> 
> @3v1n0 Ehm... I did it by adding `this.setProgress(0.4);` at the end of the constructor... 😅

Cool, thank you!

IMO the highlight could be removed too. It is a bit hard to tell in this screenshot but I think there is a soft highlight at the edges of the bar? :) 👓   👴

**Comment by sergio-costas** [comment]
That's why we have Gimp :-D

![imagen](https://github.com/ubuntu/yaru/assets/105285003/977c1d90-24b0-4920-be49-1abb73c7e1b0)

**Comment by sergio-costas** [comment]
@Feichtmeier There is a border both in the track and the bar. I can remove both, if you want.

@3v1n0 Should I better keep the original code that allows a gradient, and simply add extra configurable parameters in the SCSS to allow to set the colors as desired here, without a gradient?

**Comment by sergio-costas** [comment]
@Feichtmeier The Mac has borders in both, too

![imagen](https://github.com/ubuntu/yaru/assets/105285003/4da7c4dc-4e3f-47bd-9d7c-ee5492199abd)

**Comment by 3v1n0** [comment]
> @3v1n0 Should I better keep the original code that allows a gradient, and simply add extra configurable parameters in the SCSS to allow to set the colors as desired here, without a gradient?

Yeah, if you manage to keep both ways is fine, since there's people who customize the theme too

**Comment by Feichtmeier** [comment]
> @Feichtmeier There is a border both in the track and the bar. I can remove both, if you want.
> 
> @3v1n0 Should I better keep the original code that allows a gradient, and simply add extra configurable parameters in the SCSS to allow to set the colors as desired here, without a gradient?

the parameter would be the best I think!

Yes the mac has it too, the example was not perfect.
We should better look around at other OSD-styled controls in the gnome shell. I can't see any element there anymore which looks this "engraved" looks :) 
Ofc we could get more opinions in if you like. But I know that those time windows where we can adapt yaru are pretty small

**Comment by sergio-costas** [comment]
Maybe using a darker color...

![imagen](https://github.com/ubuntu/yaru/assets/105285003/6e20176d-47a8-4b58-a738-0b478fc8e6b0)

**Comment by Feichtmeier** [comment]
The darker color is nice but IMO I would try to remove the white outline on the bar, and just make the whole bar white without the grey in the middle

**Comment by sergio-costas** [comment]
@Feichtmeier @3v1n0 Ok, I did several changes and now it is possible to define ALL the progress bar parameters using CSS. It also, by default, keeps the current style. Can you review the patch, please? When it is merged, I will prepare a patch for Yaru that changes the style to what you want.

**Comment by Feichtmeier** [comment]
> Can you review the patch, please? When it is merged, I will prepare a patch for Yaru that changes the style to what you want.

Thank you very much for your work!
I can not review the GJS code though as I never built gnome shell myself :(

**Comment by Feichtmeier** [comment]
Apart from the "dated" discussion, now the styling is broken:

<img width="480" height="270" alt="Image" src="https://github.com/user-attachments/assets/256a9be1-4a9e-4d8a-9e24-517d11844f2d" />
